pkg/observer/controller.go

```diff
@@ -334,12 +380,31 @@ func (c *controller) Run() {
 		}
 	}
 
-	// wait for the desired amount to allow any in progress changes to batch up
+    // Filter to only the lowest priority nodegroups
+    lowestPriorityBatch := c.selectLowestPriorityNodeGroups(changedNodeGroups)
+    if len(lowestPriorityBatch) == 0 {
+        klog.V(2).Infoln("no nodegroups to apply after priority filter")
+        return
+    }
+
+    // If any lower priority CNRs are still in progress, skip this run
+    batchPriority := max(lowestPriorityBatch[0].NodeGroup.Spec.Priority, 0)
+    if c.hasLowerPriorityCNRsInProgress(batchPriority, inProgressCNRs) {
+        c.BlockedNodeGroups.WithLabelValues().Set(float64(len(lowestPriorityBatch)))
+        klog.V(2).Infof("lower priority CNRs still in progress for priority < %d; skipping creation", batchPriority)
+        return
+    }
+    // Not blocked
+    c.BlockedNodeGroups.WithLabelValues().Set(0)
+
+    // wait for the desired amount to allow any in progress changes to batch up
 	klog.V(3).Infof("waiting for %v to allow changes to settle", c.WaitInterval)
 	select {
-	case <-time.After(c.WaitInterval):
-		klog.V(3).Infof("applying %d CNRs", len(changedNodeGroups))
-		c.createCNRs(changedNodeGroups)
+    case <-time.After(c.WaitInterval):
+        klog.V(3).Infof("applying %d CNRs (lowest priority batch)", len(lowestPriorityBatch))
+        c.ApplyingNodeGroups.WithLabelValues().Set(float64(len(lowestPriorityBatch)))
+        c.createCNRs(lowestPriorityBatch)
+        c.ApplyingNodeGroups.WithLabelValues().Set(0)
```

Comment:
issue: setting this immediately back to 0 will overwrite the metric you set above if we are scraping via prometheus, as the metric is only captured every scrape interval (e.g. 15s or 30s). I think the way to handle this situation is to make the metrics counters, and increment them by the number that we applied. Then we can use `rate()` functions to view the changes.